src/CodeGen_X86.cpp

```diff
@@ -167,25 +168,34 @@ bool should_use_pmaddwd(const Expr &a, const Expr &b, vector<Expr> &result) {
     Type t = a.type();
     internal_assert(b.type() == t);
 
-    const Mul *ma = a.as<Mul>();
-    const Mul *mb = b.as<Mul>();
-
-    if (!(ma && mb && t.is_int() && t.bits() == 32 && (t.lanes() >= 4))) {
+    if (!(t.is_int() && t.bits() == 32 && (t.lanes() >= 4))) {
```

Comment:
nit: The extra parens around `t.lanes() >= 4` is unnecessary and hampers readability